README.md

```diff
@@ -1,59 +1,199 @@
 
 ## Installation
 
 ```sh
-$ npm install @moxy/{package-name}
+$ npm install @moxy/react-navigation
 ```
 
 This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
 
 ## Motivation
 
-{package-motivation}
+Most websites have a navbar and a drawer to provide an easy access to other pages. The creation process is usually remade project after project with little to no reutilization.
 
-## Usage
+By analysing a few projects, most of the logic was identical in every single one. This raised a few questions and one solution to the problem would be to have a set of react tools to simplify the process.
 
-{package-usage-example}
+A concern that came from the proposed solution was that every single website has different designs for a navbar or a drawer. Although this is true, their behaviour is usually the same. So, the solution should only deal with logic and behaviours while giving total freedom of the content that is rendered.
 
-## API
+## Usage
 
-{package-api-description}
+```js
+import React from 'react';
+import { NavigationProvider, Navbar, Drawer, useNavigation } from '@moxy/react-navigation';
+
+const MyNavigationHelper = () => {
+    const { drawer } = useNavigation();
+    
+    return (
+        <>
+            <span>{ `Is Drawer Open?  ${drawer.isOpen}` }</span>
+            <button onClick={ drawer.open }>Open Drawer</button>
+            <button onClick={ drawer.close }>Close Drawer</button>
+            <button onClick={ drawer.toggle }>Toggle Drawer</button>
+        </>
+    );
+}
+
+const App = ({ Page, pageProps }) => (
+    <NavigationProvider>
+        <Navbar
+            placement="top"
+            behaviour="hideOnScrollDown"
+            className"...">
+            <p>This is the content of the navbar.</p>
+            <MyNavigationHelper />
+        </Navbar>
+        <Drawer
+            withOverlay
+            placement="left"
+            className="..."
+            overlayClassName="...">
+            <p>This is the content of the navbar.</p>
+        </Drawer>
+        <Page { ...pageProps } />
+    </NavigationProvider>
+);
+
+export default App;
+```
 
-#### {package-api-prop-example}
+Import the styleguide base styles in the app's entry CSS file:
+
+```css
+@import "@moxy/react-navigation/dist/index.css";
+```
 
-Type: `object`
-Required: `true`
+..or in your entry JavaScript file:
 
-The `{package-api-prop-example}` has the following shape:
 ```js
-{package-api-prop-example}: PropTypes.shape({
-    foo: PropTypes.string,
-    bar: PropTypes.arrayOf(PropTypes.object),
-}).isRequired,
+import '@moxy/react-navigation/dist/index.css';
 ```
 
+## Styling
+
+Each provided component accepts classNames for styling. Although there might be a need to style conditionally. Some auxiliary data attributes are provided and can be styled like so:
+
+```css
+.drawer,
+.drawer[data-open="false"] {
```

Comment:
I loved the addition of this data attribute

Reply:
The placement one is also really good

Reply:
I really don't like having components with a lot of classNames for stuff like:
`openDrawerClassName`
`closeDrawerClassName`
`openDrawerOverlayClassName`
`closeDrawerOverlayClassName`

Data attributes gives a lot of freedom in both styling and many more.

Reply:
It's a good way to make your component's API way cleaner 👍